cirq/protocols/resolve_parameters.py

```diff
@@ -47,7 +47,8 @@ def _parameter_names_(self: Any) -> AbstractSet[str]:
         """
 
     @doc_private
-    def _resolve_parameters_(self: Any, param_resolver: 'cirq.ParamResolver'):
+    def _resolve_parameters_(self: Any, param_resolver: 'cirq.ParamResolver',
```

Comment:
It is possible to express the fact that the function takes Any type, but returns the same type using [generic functions](https://mypy.readthedocs.io/en/stable/generics.html#generic-functions).

(Also applies to `resolve_parameters_once` elsewhere.)

Reply:
Unfortunately this generates mypy errors, since raw Symbols can resolve to floats while other types expect the input type back. `Union[float, BaseType]` gives errors in places that explicitly require `BaseType` back.

I don't intend to address this in this PR since I've already rabbit-holed away from subcircuits a fair amount, but please open an issue if you'd like to see this resolved.

Reply:
SGTM!